tests/components/youless/test_sensor.py

```diff
@@ -0,0 +1,116 @@
+"""Test the sensor classes for youless."""
+
+from unittest.mock import MagicMock
+
+import youless_api
+
+from homeassistant.components.sensor import SensorStateClass
+from homeassistant.components.youless.sensor import (
+    CurrentPowerSensor,
+    DeliveryMeterSensor,
+    EnergyMeterSensor,
+    GasSensor,
+    WaterSensor,
+)
+
+
+def test_water_sensor():
+    """Check the water sensor setup."""
+    mock_coordinator = MagicMock(
+        data=MagicMock(model="LS120", water_meter=youless_api.YoulessSensor(123.12, ""))
+    )
+
+    water_sensor = WaterSensor(mock_coordinator, "localhost")
+
+    assert water_sensor.unique_id == "youless_localhost_water"
+    assert water_sensor.device_info.get("manufacturer") == "YouLess"
+    assert water_sensor.device_info.get("model") == "LS120"
+    assert water_sensor.device_info.get("name") == "Water meter"
+    assert water_sensor.name == "Water usage"
+    assert water_sensor.native_value == 123.12
+    assert water_sensor.available
```

Comment:
You should write tests that use the state machine, you shouldn't directly access the different classes